processing/src/main/java/org/apache/druid/query/BaseQuery.java

```diff
@@ -270,14 +270,13 @@ public boolean equals(Object o)
            Objects.equals(dataSource, baseQuery.dataSource) &&
            Objects.equals(context, baseQuery.context) &&
            Objects.equals(querySegmentSpec, baseQuery.querySegmentSpec) &&
-           Objects.equals(duration, baseQuery.duration) &&
+           Objects.equals(getDuration(), baseQuery.getDuration()) &&
            Objects.equals(granularity, baseQuery.granularity);
   }
 
   @Override
   public int hashCode()
   {
-
-    return Objects.hash(dataSource, descending, context, querySegmentSpec, duration, granularity);
```

Comment:
curious about why this change. I know this isn't really part of your change, but I've also noticed equals and hashcode for a lot of classes that extend BaseQuery have their own implementation of equals and hashCode and do not check the base class. Sounds like we should add some equalsVerifier tests for them, maybe in another patch.

Reply:
It's because `duration` is lazily computed, so it might be null. Calling `getDuration()` forces it to be computed. I'll add a comment.

Btw, a lot of the subclasses of BaseQuery _do_ call `super.equals` at some point (but they do other stuff too).

Reply:
Weird  - I see the calls to `super.equals()` now, maybe I just shouldn't read code in the night 😝 

EqualsVerifier is still complaining about some other stuff. But cleaning that up can be another issue.